### ISSUE TITLE
Exception handling

### ISSUE THREAD

**Issue by sridhard** [mozilla/pdf.js#11469]
Attach (recommended) or Link to PDF file here:
_Happens for any PDF file_

Configuration:
Web browser and its version: Chrome -79
Operating system and its version: MacOS-Mojave
PDF.js version:v2.3.200
Is a browser extension: No

Steps to reproduce the problem: 
_I am getting the below crash when we zoom or rotate pages faster (faster means before zoom is completed zoom again, continuous clicking)_

```
pdf.js:20157 Uncaught (in promise) Error: TextLayer task cancelled.
    at TextLayerRenderTask.TextLayer_cancel [as cancel] (pdf.js:20157)
    at TextLayerBuilder.cancel (pdf_viewer.js:2855)
    at PDFPageView.cancelRendering (pdf_viewer.js:6146)
    at PDFPageView.reset (pdf_viewer.js:6027)
    at PDFPageView.update (pdf_viewer.js:6131)
    at PDFViewer.set (pdf_viewer.js:8049)
    at Object.rotatePages (pdfApp.js:277)
```

_1. I am not able to catch the above exception
2. Can you please let know how to solve the crash. I think the exception is ok because it is trying to cancel the previous render, please correct me if I am wrong._

**_Note: I am not using the default PDF viewer. I am using PDFJS api to build my own viewer._**

What is the expected behavior? (add screenshot)
There should not be any exception

What went wrong? (add screenshot)
There is exception and not able to catch exception

Link to a viewer (if hosted on a site other than mozilla.github.io/pdf.js or as Firefox/Chrome extension):
It is not yet hosted. I am testing in my local machine.

**Comment by Snuffleupagus** [comment]
When opening an issue, you need to provide *all* of the information requested in https://github.com/mozilla/pdf.js/blob/master/.github/ISSUE_TEMPLATE.md

**Comment by sridhard** [comment]
@Snuffleupagus I updated the information as per template

**Comment by Snuffleupagus** [comment]
> ```
> pdf.js:20157 Uncaught (in promise) Error: TextLayer task cancelled.
> ...
> ```
That message simply informs you that there was a `catch` missing in a promise-chain, but doesn't *necessarily* mean that there's any breaking errors.

> [...] I think the exception is ok because it is trying to cancel the previous render, 

This sentence seem to suggest that there's no *actual* errors, but rather only the console message as quoted, which makes the use of words such as "crash" and "exception" in https://github.com/mozilla/pdf.js/issues/11469#issue-544990393 slightly difficult to interpret (since those would generally only be used if things break completely).

> PDF.js version: v2.3.200

The only case of "Uncaught promise" messages, relevant for the `TextLayer` code, I'm able to recall was fixed by PR #10962, which is included in that version of the library; see https://github.com/mozilla/pdf.js/releases/tag/v2.3.200

Hence this issue unfortunately doesn't really make sense, given the currently provided information.

> It is not yet hosted. I am testing in my local machine.

Considering everything above, the only way to make this issue actionable would be actually provide a *runnable* example, please see https://github.com/mozilla/pdf.js/blob/master/.github/CONTRIBUTING.md (emphasis mine):
> If you are developing a custom solution, first check the examples at https://github.com/mozilla/pdf.js#learning and search existing issues. If this does not help, **please prepare a short well-documented example that demonstrates the problem and make it accessible online** on your website, JS Bin, GitHub, etc. before opening a new issue or contacting us on the IRC channel -- **keep in mind that just code snippets won't help us troubleshoot the problem**.

**Comment by sridhard** [comment]
ok..thanks for your help.
